test-container/src/main/java/io/strimzi/test/container/StrimziKafkaCluster.java

```diff
@@ -0,0 +1,144 @@
+        defaultKafkaConfigurationForMultiNode.put("num.partitions", String.valueOf(internalTopicReplicationFactor));
+        defaultKafkaConfigurationForMultiNode.put("transaction.state.log.replication.factor", String.valueOf(internalTopicReplicationFactor));
+        defaultKafkaConfigurationForMultiNode.put("transaction.state.log.min.isr", String.valueOf(internalTopicReplicationFactor));
+
+        additionalKafkaConfiguration.putAll(defaultKafkaConfigurationForMultiNode);
+
+        // multi-node set up
+        this.brokers = IntStream
+            .range(0, this.brokersNum)
+            .mapToObj(brokerId -> {
+                LOGGER.info("Starting broker with id {}", brokerId);
+                // adding broker id for each kafka container
+                return StrimziKafkaContainer.createWithAdditionalConfiguration(brokerId, imageVersion, additionalKafkaConfiguration)
+                    .withNetwork(this.network)
+                    .withNetworkAliases("broker-" + brokerId)
+                    .dependsOn(this.zookeeper)
+                    .withExternalZookeeper("zookeeper:" + StrimziZookeeperContainer.ZOOKEEPER_PORT);
+            })
+            .collect(Collectors.toList());
+    }
+
+    public StrimziKafkaCluster(Map<String, String> additionalKafkaConfiguration) {
+        this(StrimziKafkaContainer.getStrimziVersion() + "-kafka-" + StrimziKafkaContainer.getLatestKafkaVersion(),
+            3,
+            3,
+            additionalKafkaConfiguration);
+    }
+
+    public Collection<StrimziKafkaContainer> getBrokers() {
+        return this.brokers;
+    }
+
+    public String getBootstrapServers() {
+        return brokers.stream()
+            .map(StrimziKafkaContainer::getBootstrapServers)
+            .collect(Collectors.joining(","));
+    }
+
+    @Override
+    public void start() {
+        Stream<? extends Startable> startables = this.brokers.stream();
+        try {
+            Startables.deepStart(startables).get(60, TimeUnit.SECONDS);
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            e.printStackTrace();
+        }
+
+        TestUtils.waitFor("Broker node", Duration.ofSeconds(1).toMillis(), Duration.ofSeconds(30).toMillis(),
+            () -> {
+                Container.ExecResult result = null;
+                try {
+                    result = this.zookeeper.execInContainer(
+                        "sh", "-c",
+                        "bin/zookeeper-shell.sh zookeeper:" + StrimziKafkaContainer.ZOOKEEPER_PORT + " ls /brokers/ids | tail -n 1"
+                    );
+                    String brokers = result.getStdout();
+
+                    LOGGER.info("Stdout from zookeeper container....{}", result.getStdout());
+
+                    return brokers != null && brokers.split(",").length == this.brokersNum;
+                } catch (IOException | InterruptedException e) {
+                    e.printStackTrace();
```

Comment:
Remind me again why it's OK to just ignore the `InterruptedException`? I know it's an annoying Exception to have to deal with, and I know you're not expecting to need to support cancellation here, but you never know how code will evolve and be reused. It would be better to do `Thread.currentThread().interrupt();`, that way `waitFor`'s call to `sleep()` would throw `InterruptedException` and we can declare `waitFor() throws InterruptedException`, and then the whole thing supports cancellation.